account_financial_discount/models/account_reconcile_model.py

```diff
@@ -0,0 +1,282 @@
+                        "A financial discount expense account must be set on "
+                        "the company"
+                    )
+                )
+            if errors:
+                msg = (
+                    _(
+                        "Reconciliation model %s is set to consider financial "
+                        "discount. However to function properly:\n"
+                    )
+                    % rec.name
+                )
+                raise ValidationError(msg + " - " + "\n - ".join(errors))
+
+    def _get_write_off_move_lines_dict(self, st_line, move_lines=None):
+        """Prepare financial discount write-off"""
+        if self.rule_type != "invoice_matching" or not self.apply_financial_discounts:
+            return super()._get_write_off_move_lines_dict(
+                st_line, move_lines=move_lines
+            )
+        # Copied from odoo
+        line_residual = (
+            st_line.currency_id and st_line.amount_currency or st_line.amount
+        )
+        line_currency = (
+            st_line.currency_id
+            or st_line.journal_id.currency_id
+            or st_line.company_id.currency_id
+        )
+        total_residual = (
+            move_lines
+            and sum(
+                aml.currency_id and aml.amount_residual_currency or aml.amount_residual
+                for aml in move_lines
+            )
+            or 0.0
+        )
+        balance = total_residual - line_residual
+
+        if float_is_zero(balance, precision_rounding=line_currency.rounding):
+            return []
+
+        discount = sum(move_lines.mapped("amount_discount"))
+
+        write_off_account = (
+            self.financial_discount_expense_account_id
+            if discount > 0
+            else self.financial_discount_revenue_account_id
+        )
+        fin_disc_write_off_vals = {
+            "name": self.financial_discount_label,
+            "account_id": write_off_account.id,
+            "debit": discount > 0 and discount or 0,
+            "credit": discount < 0 and -discount or 0,
+            "reconcile_model_id": self.id,
+        }
+        tax_discount = sum(move_lines.mapped("amount_discount_tax"))
+        if not tax_discount:
+            return [fin_disc_write_off_vals]
+        res = []
+        for line in move_lines:
+            tax_line = line.discount_tax_line_id
+            if not tax_line:
+                continue
+            tax_write_off_vals = {
+                "name": tax_line.name,
+                "account_id": tax_line.account_id.id,
+                "debit": tax_line.credit and line.amount_discount_tax or 0,
+                "credit": tax_line.debit and line.amount_discount_tax or 0,
+                "reconcile_model_id": self.id,
+            }
+            # Deduce tax amount from fin. disc. write-off
+            if fin_disc_write_off_vals.get("credit"):
+                fin_disc_write_off_vals["credit"] -= line.amount_discount_tax
```

Comment:
```suggestion
                fin_disc_write_off_vals["credit"] += line.amount_discount_tax
```
While working on mig to v14.0 I'm wondering if we shouldn't have this here (for vendor bills)